### PR TITLE
Fix merge validation

### LINKED ISSUE #41: Merge Validation Failing
It appears that when two names are merged with each other, the ValidationError is not being raised. However when three names are merged in the same fashion, the ValidationError _is_ raised.

Fix the validation to prevent two names from being merged with each other.

### PR DESCRIPTION
**Pull request by damonkelley**
Resolves #41 

It seems that the reason this was failing is some scenarios because in order to properly validate the FK, we needed have the target Name and the Name being merged. _Normal_ validators are only passed the ID of the related model for ForeignKey fields. 

So...

I modified `validate_merged_with` to accept a Name instance in order to properly validate with itself and the merge target, and it is now being called from the `clean` method on the Name model.

P.S. There is an issue with TravisCI ATM so the builds are not as reliable here. There is something going on with the ability to pick up env variables which is not allowing the tests to connect to the database.

### CODE REVIEW

**Comment by damonkelley** [comment]
ping @vphill @ldko @somexpert 

This is ready for review!

**Comment by ldko** [comment]
+1

File: tests/test_validators.py [hunk 1]
```diff
@@ -1,23 +1,34 @@
 import pytest
-from name.models import validate_merged_with, Name
+from name.validators import validate_merged_with
```
Comment (somexpert) [inline tests/test_validators.py]: These imports should be grouped by type and separated by a blank line according to PEP-8 standards

Comment (damonkelley) [inline tests/test_validators.py]: Good catch @somexpert!

File: tests/test_validators.py [hunk 2]
```diff
@@ -45,13 +57,53 @@ def test_validate_merged_with_fails_when_name_merges_with_itself():
...
+
+    john.merged_with = ben
+    john.save()
+
+    validate_merged_with(john)
+
+
+@pytest.mark.django_db
+def test_validate_merged_with_when_name_changed_merged_with_to_invalid_name():
+    """Check that validation if a valid merged_with related model is
```
Comment (somexpert) [inline tests/test_validators.py]: I could be wrong, but should the 'if' be an 'of'?

**Comment by somexpert** [comment]
:thumbsup:

**Pull request merged**